R/estimateDominance.R

```
@@ -0,0 +1,366 @@
+#' #Shows all indices
+#' colData(esophagus)
+#' #Shows DBP index
+#' colData(esophagus)$dbp
+#'
+#' #Deletes DBP index
+#' colData(esophagus)$dbp <- NULL
+#' #Shows all indices, DBP is deleted
+#' colData(esophagus)
+#' #Deletes all indices
+#' colData(esophagus) <- NULL
+#'
+#' #Names of columns can be chosen, but the length of arguments must match.
+#' esophagus <- estimateDominance(esophagus,
+#'     index=c("dbp", "core_abundance"),
+#'     name = c("index1", "index2"))
+#' #Shows all indices
+#' colData(esophagus)
+#' #If they do not match, gets an error.
+#' esophagus <- estimateDominance(esophagus,
+#'     index="simpson",
+#'     name = c("index3", "index4"))
+#' #Shows all indices
+#' colData(esophagus)
+#' #Deletes all indices
+#' colData(esophagus) <- NULL
+#'
+#' #Calculates all indices
+#' esophagus <- estimateDominance(esophagus)
+#' #Or you can calculate all indices by specify "all"
+#' esophagus <- estimateDominance(esophagus, index="all")
+#' #Shows all indices
+#' colData(esophagus)
+#'
+#' @export
+#'
+setGeneric("estimateDominance",signature = c("x"),
+           function(x, abund_values = "counts", index="all", rank=1, as_relative=TRUE, aggregate=TRUE,
+                    name = index, ...)
+               standardGeneric("estimateDominance"))
+
+
+#' @rdname estimateDominance
+#' @export
+setMethod("estimateDominance", signature = c(x = "MicrobiomeExperiment"),
+          function(x, abund_values = "counts", index="all", rank=1, as_relative=TRUE, aggregate=TRUE,
+                   name = index, BPPARAM = SerialParam(), ...){
+
+              #Check if index/indices that user wants to calculate are accepted. Those that are, are stored in "index" variable.
+              index <- checkIndices(index)
+
+              #Initialize table that is used to store indices
+              tab <- NULL
+
+              #If the indices list is not null
+              if (!is.null(index)) {
```

If `index` is `NULL`, something has gone wrong. The user should not get away with it. "fail early, fail often"

---

Corrected